src/Microsoft.SqlTools.Authentication/AuthenticationParams.cs

```diff
@@ -0,0 +1,64 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.SqlTools.Authentication
+{
+    /// <summary>
+    /// Parameters to be passed to <see cref="Authenticator"/> to request an access token
+    /// </summary>
+    public class AuthenticationParams
+    {
+        /// <summary>
+        /// Authentication method to be used by <see cref="Authenticator"/>.
+        /// </summary>
+        public AuthenticationMethod AuthenticationMethod { get; set; }
+
+        /// <summary>
+        /// Authority URL, e.g. https://login.microsoftonline.com/
+        /// </summary>
+        public string Authority { get; set; }
+
+        /// <summary>
+        /// Audience for which access token should be acquired, e.g. common, organizations, consumers.
+        /// It can also be a tenant Id when authenticating multi-tenant application accounts.
+        /// </summary>
+        public string Audience { get; set; }
+
+        /// <summary>
+        /// Array of scopes for which access token is requested.
+        /// </summary>
+        public string[] Scopes { get; set; }
+
+        /// <summary>
+        /// <see cref="Guid"/> Connection Id, that will be passed to Azure AD when requesting access token.
+        /// It can be used for tracking AAD request status if needed.
+        /// </summary>
+        public Guid ConnectionId { get; set; }
+
+        /// <summary>
+        /// User name to be provided as userhint when acquiring access token.
+        /// </summary>
+        public string Username { get; set; }
```

Comment:
I think most other places in this project we can this `UserName`.  Maybe better to be consistent if there isn't a good reason not to be?